src/mongoChangeStream.test.ts

```diff
@@ -42,6 +42,63 @@ const getSync = async (options?: SyncOptions) => {
   return sync
 }
 
+/**
+ * Repeatedly checks the provided predicate until it either returns true or the
+ * timeout is reached.
+ *
+ * @param pred - The predicate to check: an async function returning a boolean.
+ * @param timeout - The maximum time to wait for the predicate to return true.
+ * @param interval - The time to wait between checks.
+ *
+ * @returns true if the predicate returned true before the timeout, false
+ * otherwise.
+ */
+const eventually = async (
+  pred: () => Promise<boolean>,
+  timeout: number,
+  interval: number
+) => {
+  const start = Date.now()
+
+  while (Date.now() - start < timeout) {
+    if (await pred()) {
+      return true
+    }
+    await setTimeout(interval)
+  }
```

Comment:
I could probably do this more elegantly using `Promise.race`. Then the interval wouldn't matter.

Reply:
Although maybe not... I'd still need a `while` loop to repeatedly check the predicate with a sleep in between.

Reply:
There is already a function that does this in `prom-utils`. See `waitUntil` https://www.npmjs.com/package/prom-utils.

Reply:
BTW, `raceTimeout` handles the other scenario you were considering earlier in this comment thread.

Reply:
Let's remove this function and use `waitUntil`.

Reply:
Nice! Will do.